lib/slots.ts

```diff
@@ -17,37 +19,41 @@ export type GetSlots = {
 
 const getMinuteOffset = (date: Dayjs, step: number) => {
   // Diffs the current time with the given date and iff same day; (handled by 1440) - return difference; otherwise 0
-  const minuteOffset = Math.min(date.diff(dayjs().startOf("day"), "minutes"), 1440) % 1440;
+  const minuteOffset = Math.min(date.diff(dayjs().startOf("day"), "minute"), 1440) % 1440;
   // round down to nearest step
-  return Math.floor(minuteOffset / step) * step;
+  return Math.ceil(minuteOffset / step) * step;
```

Comment:
If the time was 13:37 flooring caused the last past slot to be available, ceil to resolve.